### PR TITLE
[self-hosted][docs] Fix some typos, dead links missing information

### PR DESCRIPTION
**Pull request by corneliusludmann**
There are some dead links and typos in the self-hosted documentation that this commit fixes. It also add some minor additions that I found useful to know while testing Gitpod self-hosted.

### CODE REVIEW

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md [hunk 1]
```diff
@@ -5,19 +5,18 @@ However, Gitpod sports a range of integrations with the Google Cloud Platform th
 This section describes the neccesary steps to prepare a new GCP project for a Gitpod installation.
 
   > We have automated this process. If you want to get up and running as quickly as possible, that's the way to go.
-  > See the [automated setup](#automated-setup) section for details.
 
-You can perform the set up of the GCP resources yourself. See the [manual GCP setup](12_install_on_gcp_manual) section for more details.
+You can perform the set up of the GCP resources yourself. See the [manual GCP setup](../12_install_on_gcp_manual) section for more details.
```
Comment (jankeromnes) [inline src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: Shouldn't there be a trailing `/` at the end? (I believe @32leaves fixed a few broken links that way recently.)

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md [hunk 2]
```diff
@@ -5,19 +5,18 @@ However, Gitpod sports a range of integrations with the Google Cloud Platform th
...
 
 This installation script configures your GCP project and produces the required Helm configuration for installing Gitpod in that project.
-Once all GCP setup is complete, you will only have to manually set up [OAuth](30_oauth) and optionally [HTTPS](34_https_certs).
+Once all GCP setup is complete, you will only have to manually set up [OAuth](../30_oauth) and optionally [HTTPS](../34_https_certs).
```
Comment (jankeromnes) [inline src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: Same as above.

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md [hunk 3]
```diff
@@ -26,14 +25,16 @@ You can install Gitpod self-hosted in your GCP project from within Gitpod. This
...
-We recommend you fork this repository so that you can easily rebase your changes on the latest version.
+We recommend you to fork this repository so that you can easily rebase your changes on the latest version.
 
 ```
 git clone https://github.com/gitpod-io/self-hosted
 cd self-hosted
 git remote rename origin upstream
 ```
 
+Configure at least `values.yaml`. Set your domain and the [OAuth](../30_oauth) section.
```
Comment (jankeromnes) [inline src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: Trailing `/`?

Comment (jankeromnes) [reply src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: Suggestion:

```suggestion
Configure at least your `values.yaml`. Set your domain and the [OAuth](../30_oauth) section.
```

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md [hunk 4]
```diff
@@ -5,19 +5,18 @@ However, Gitpod sports a range of integrations with the Google Cloud Platform th
 This section describes the neccesary steps to prepare a new GCP project for a Gitpod installation.
 
   > We have automated this process. If you want to get up and running as quickly as possible, that's the way to go.
-  > See the [automated setup](#automated-setup) section for details.
```
Comment (jankeromnes) [inline src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: This no longer exists, right?

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md (hunk 1 above)
Comment (corneliusludmann) [reply src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: In my tests it works with and without trailing `/` as it is added automatically. I saw some other links without trailing `/` as well. However, I could add them to be save …

File: src/docs/self-hosted/0.3.0/install/35_docker_registry.md [hunk 5]
```diff
@@ -1,10 +1,10 @@
 # Docker Registry
 
-Gitpod uses a Docker registry to push the workspace images it builds. 
-This helm chart can either deploy its own registry (default but requires HTTPS certs, see above) or use an existing one. 
+Gitpod uses a Docker registry to push the workspace images it builds.
+This helm chart can either deploy its own registry (default but requires [HTTPS certs](../34_https_certs)) or use an existing one.
```
Comment (jankeromnes) [inline src/docs/self-hosted/0.3.0/install/35_docker_registry.md]: Trailing `/`? (I'm actually not sure, feel free to dismiss all these comments if the links work in Netlify).

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md (hunk 1 above)
Comment (jankeromnes) [reply src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: As long as it works on the live Netlify website it's cool. 🙂

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md (hunk 4 above)
Comment (corneliusludmann) [reply src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: The handle `#automated-setup` does not exist. I have no idea where the links should point to instead.

File: src/docs/self-hosted/0.3.0/install/35_docker_registry.md (hunk 5 above)
Comment (32leaves) [reply src/docs/self-hosted/0.3.0/install/35_docker_registry.md]: Trailing `/` would be great

File: src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md (hunk 4 above)
Comment (32leaves) [reply src/docs/self-hosted/0.3.0/install/11_install_on_gcp_script.md]: The anchor used to be created by a corresponding headline. That one's gone by now.
Removing this line is the best choice.

**Pull request merged**